shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -521,12 +522,14 @@ public void writeFields(DataOutput out) throws IOException {
             out.writeLong(writerId.getMostSignificantBits());
             out.writeLong(writerId.getLeastSignificantBits());
             out.writeLong(eventNumber);
+            out.writeLong(previousEventNumber);
         }
 
         public static WireCommand readFrom(DataInput in, int length) throws IOException {
             UUID writerId = new UUID(in.readLong(), in.readLong());
             long offset = in.readLong();
-            return new DataAppended(writerId, offset);
+            long previousEventNumber = in.readLong();
```

Comment:
This is not compatable. If a new client reads data from an old server it will get an exception here.

Reply:
We haven't actually agreed whether we support a new client talking to an old server. We haven't discarded the option of not supporting such a scenario as upgrades in live systems typically prefer to upgrade servers first. The decision on what to do here will depend on the outcome of [PDP 21](https://github.com/pravega/pravega/wiki/PDP-21:-Protocol-revisioning).

Reply:
ok... but this does not comply with the rest of the PDP either. It has not incremented the version number, and if you want to make it backward compatible, the versions sent by the client and the server in the Hello need to be changed to exclude the old version.

Reply:
Let's agree about what to do here. There are three options I see:

1. I reintroduce the length check, and maybe bump up the current version
2. I bump up both current version and old version to say that previous versions are incompatible
3. I leave as is

Under the PDP guidelines, I believe we need to do either 1 or 2. Given that no one really needs to upgrade Pravega at this point, I'd say that it is fine to do 3, which I understand makes it incompatible, but given that all current applications are currently testing, they should really upgrade immediately. Doing 1 or 2 right now might be a good idea so that we can start getting the gist of making such changes.

Do you have any preference? I don't have any strong preference at this point.

Reply:
I'd prefer to do 1. We can easily transition to removing the length check at any future date. 
version bump: per the PDP we should, but I'm not sure it is required, especially if we have a length check.